GAE/src/org/akvo/flow/xml/XmlQuestion.java

```diff
@@ -56,28 +60,171 @@
     @JacksonXmlProperty(localName = "mandatory", isAttribute = true)
     private boolean mandatory;
     @JacksonXmlProperty(localName = "requireDoubleEntry", isAttribute = true)
-    private boolean requireDoubleEntry;
+    private Boolean requireDoubleEntry;
     @JacksonXmlProperty(localName = "localeNameFlag", isAttribute = true)
     private boolean localeNameFlag;
     @JacksonXmlProperty(localName = "locked", isAttribute = true)
-    private boolean locked;
+    private Boolean locked;
     @JacksonXmlProperty(localName = "localeLocationFlag", isAttribute = true)
-    private boolean localeLocationFlag;
+    private Boolean localeLocationFlag;
     @JacksonXmlProperty(localName = "caddisflyResourceUuid", isAttribute = true)
     private String caddisflyResourceUuid;
     @JacksonXmlProperty(localName = "cascadeResource", isAttribute = true)
     private String cascadeResource;
     @JacksonXmlProperty(localName = "allowPoints", isAttribute = true)
-    private boolean allowPoints;
+    private Boolean allowPoints;
     @JacksonXmlProperty(localName = "allowLine", isAttribute = true)
-    private boolean allowLine;
+    private Boolean allowLine;
     @JacksonXmlProperty(localName = "allowPolygon", isAttribute = true)
-    private boolean allowPolygon;
+    private Boolean allowPolygon;
 
 
     public XmlQuestion() {
     }
 
+    //Create a jackson object from a domain object
+    public XmlQuestion(Question q) {
+        text = q.getText();
+        id = q.getKey().getId();
+        order = q.getOrder();
+        mandatory = Boolean.TRUE == q.getMandatoryFlag();
```

Comment:
```suggestion
        mandatory = Boolean.TRUE.equals(q.getMandatoryFlag());
```

Reply:
I would suggest keeping the convention across the rest of the boolean values below.